lms/djangoapps/django_comment_client/base/views.py

```diff
@@ -155,6 +176,82 @@ def create_thread(request, course_id, commentable_id):
         return JsonResponse(safe_content(data, course_key))
 
 
+def _send_discussion_notification(
+    type_name,
+    course_id,
+    thread,
+    request_user,
+    recipient_user_id=None,
+    recipient_group_id=None,
+    recipient_exclude_user_ids=None,
+    extra_payload=None,
+):
+    """
+    Helper method to consolidate Notification trigger workflow
+    """
+    try:
+        # is Notifications feature enabled?
+        if not settings.FEATURES.get("ENABLE_NOTIFICATIONS", False):
+            return
+
+        # get the notification type.
+        msg = NotificationMessage(
+            msg_type=get_notification_type(type_name),
+            namespace=course_id,
+            # base payload, other values will be passed in as extra_payload
+            payload={
+                '_schema_version': '1',
+                'action_username': request_user.username,
+                'thread_title': thread.title,
+            }
+        )
+
+        # add in additional payload info
+        # that might be type specific
+        if extra_payload:
+            msg.payload.update(extra_payload)
+
+        # Add information so that we can resolve
+        # click through links in the Notification
+        # rendering, typically this will be used
+        # to bring the user back to this part of
+        # the discussion forum
+
+        #
+        # IMPORTANT: This can be changed to msg.add_click_link() if we
+        # have a URL that we wish to use. In the initial use case,
+        # we need to make the link point to a different front end
+        #
+        msg.add_click_link_params({
+            'course_id': course_id,
+            'commentable_id': thread.commentable_id,
+            'thread_id': thread.id,
+        })
+
+        if recipient_user_id:
+            # send notification to single user
+            publish_notification_to_user(recipient_user_id, msg)
+
+        if recipient_group_id:
+            # Send the notification_msg to the CourseGroup via Celery
+            # But we can also exclude some users from that list
+            if settings.FEATURES.get('ENABLE_NOTIFICATIONS_CELERY', False):
+                publish_course_group_notification_task.delay(
+                    recipient_group_id,
+                    msg,
+                    exclude_user_ids=recipient_exclude_user_ids
+                )
+            else:
+                publish_course_group_notification_task(
+                    recipient_group_id,
+                    msg,
+                    exclude_user_ids=recipient_exclude_user_ids
+                )
+    except Exception, ex:
```

Comment:
Are there more specific exceptions we should be looking for here?  We have one big try wrapped around the entire workflow -- it would be better to have multiple try/excepts protecting distinct blocks of code.